src/main/java/io/github/benas/jpopulator/impl/PopulatorImpl.java

```diff
@@ -139,9 +164,24 @@ public PopulatorImpl() {
             throw new IllegalArgumentException("The number of beans to populate must be positive.");
         }
         List<Object> beans = new ArrayList<Object>();
-        for (int i = 0; i < size; i++) {
-            Object bean = populateBean(type, excludedFields);
-            beans.add(bean);
+        if (isSupportedType(type)) {
+            BeanFieldHolder beanFieldHolder = new BeanFieldHolder();
```

Comment:
I'm not sure I understand the purpose of the `BeanFieldHolder` class. What does this block do?

```java
BeanFieldHolder beanFieldHolder = new BeanFieldHolder();
Field field = null;
try {
   field = BeanFieldHolder.class.getField("field");
} catch (Exception e) {
    // this should never happen
    e.printStackTrace();
}
```